rust/storage/src/s3.rs

```diff
@@ -1035,6 +1035,22 @@ impl Configurable<StorageConfig> for S3Storage {
     ) -> Result<Self, Box<dyn ChromaError>> {
         match &config {
             StorageConfig::S3(s3_config) => {
+                let timeout_config = TimeoutConfigBuilder::default()
+                    .connect_timeout(Duration::from_millis(s3_config.connect_timeout_ms))
+                    .operation_timeout(Duration::from_millis(
+                        s3_config.request_timeout_ms * s3_config.request_retry_count as u64,
```

Comment:
Should we instead make `operation_timeout = request_timeout / retry_count` so this way we can have 3 attempts within the request_timeout. The current request_timeout semantics are for  a request with retries included and not for one attempt.